test/extended/util/framework.go

```diff
@@ -452,6 +452,9 @@ func FetchURL(url string, retryTimeout time.Duration) (response string, err erro
 			// because otherwise it's going to give up.
 			return false, nil
 		}
+		if r.StatusCode != 200 {
+			return false, fmt.Errorf("Got HTTP response %d, expected 200", r.StatusCode)
```

Comment:
I think you need to return nil here.  if you return an error, the wait.Poller is going to stop polling.

in the case that an app server returns a 404 until the app is up, you actually do want to keep polling.